src/shims/foreign_items.rs

```diff
@@ -211,6 +214,89 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
                 this.machine.static_roots.push(ptr.alloc_id);
             }
 
+            // Obtains a Miri backtrace. See the README for details.
+            "miri_get_backtrace" => {
```

Comment:
Could you please move the code in these match arms to a different file (probably a new one)? I'd like to keep `foreign_items` restricted to dispatching logic, it is big enough with just that.